Localization/drive.py

```diff
@@ -0,0 +1,248 @@
+# -*- coding: utf-8 -*-
+##@package move
+# move
+# ==============
+# Contains Robot class
+#
+#created by Frederike Duembgen, July 2015
+from __future__ import division, print_function
+import csv, time, sys, getopt, socket, os, signal
+import numpy as np
+USAGE = '''
+---------------------------------------------
+               How to use:
+---------------------------------------------
+
+python file.py [-d]
+
+the output will be saved in the "output/" folder with the name as 'driven_results.txt'
+Add -d to run program in "simulation mode" (without connecting to robot)
+
+-------------------------------------------- '''
+
+DEBUG = 0
+N = 512 # Counts per revolution
+R_wheels =10 # radius of robot wheels in mm
+D=30 # distance between two wheels in mm
+valid_array = ['i','f','b','l','r','e','v','u','d','a','o','c','sh','sa','s','pl','pr','g','tu','td','z'] # valid commands
+valid_boards = ['r','l','n','o']
+valid_params_motors = ['NOM_SPEED','KP','KI','KD','ACC_INC','ACC_DIV','MIN_SPEED','POS_MARG','SPEED_POS']
+valid_params_neck = ['NOM_PWM','NOM_ANGLE']
+valid_params_other = ['PORT','LED_GPIO','PAN_NOM_SPEED','TILT_NOM_SPEED']
+
+TCP_IP = '172.16.156.137'
+TCP_PORT = 51717
+BUFFER_SIZE = 1024
+
+def signal_handler(signal, frame):
+    ''' Interrupt handler for stopping on KeyInterrupt '''
+    print('Program stopped manually')
+    sys.exit(2)
+def touchopen(filename,*args, **kwargs):
+    ''' Open or create a file '''
+    fd = os.open(filename, os.O_RDWR | os.O_CREAT)
+    return os.fdopen(fd, *args, **kwargs)
+def get_parameters():
+    ''' Get parameters from command line '''
+    global DEBUG
+    output_result='output/output_result.txt'
+    try:
+        opts,args = getopt.getopt(sys.argv[1:],"hd",["ifile="])
+    except getopt.GetoptError:
+        print("error1", USAGE)
+        sys.exit(2)
+    for opt, arg in opts:
+        if opt in ("-h"):
+            print(USAGE)
+            sys.exit(2)
+        elif opt in ("-d"):
+            DEBUG=1
+
+    return output_result
+def read_setandget(command):
+    ''' Read p and g commands '''
+    if len(command) >= 5 : # minimum length of get and set command
+        if (command[1] == ' ') and (command[3] == ' '): # spaces after command and board
+            if valid_boards.count(command[2]) > 0: # board is valid
+                if command[0] == 'g':
+                    #Â get mode
+                    print('get mode')
+                elif command[0] == 'p':
+                    #Â set mode
+                    print('set mode')
+    return 1
+def read_file(R,inputfile):
```

Comment:
I think this function is not used. Remove it (and other unnecessary functions linked with it)